tests/test_parser.py

```diff
@@ -111,3 +111,18 @@ async def test_url_detection_multi_page_auto_routing(model_type):
             # Page 6: Detects the URL
             found = "https://github" in content
             assert found
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("depth", [1, 2])
+async def test_recursive_url_parsing(depth):
+    results = parse("https://example.com/", depth=depth)
+    assert len(results) == depth
+
+
+@pytest.mark.asyncio
+async def test_url_parsing_in_pdf():
+    sample = "examples/inputs/sample_test_doc.pdf"
+    parser_type = "AUTO"
```

Comment:
When testing the following file realized that the first page(home page, https://www.dca.ca.gov/acp/) is not parsed but a different page got parsed, did we miss something?
https://www.dca.ca.gov/acp/pdf_files/lemonlaw_qa.pdf
```
path = "../lemonlaw_qa.pdf"
results = parse(path, parser_type="AUTO", pages_per_split=1, depth=2)
```

Reply:
Seems the URL regex I'm using is faulty. It seems it's extracting the dca.ca.gov from the email address. I updated the regex and added some logic to get it to work. Not sure if there are any other edge cases.

Reply:
Checking ...

Reply:
Working now. Thanks!